AndroidXCI/lib/src/main/kotlin/dev/androidx/ci/testRunner/TestRunnerService.kt

```diff
@@ -294,6 +303,17 @@ interface TestRunnerService {
         /**
          * Test case log files produced by the test.
          */
-        val testCaseLogcats: Map<TestIdentifier, ResultFileResource>
+        val testCaseArtifacts: Map<TestIdentifier, List<TestCaseArtifact>>
     }
+
+    data class TestCaseArtifact(
+        /**
+         * Test case log files produced by individual tests in the testrun.
+         */
+        val resultFileResource: ResultFileResource,
+        /**
+         * Type of files: ex. logcat, png or textproto for screenshot tests
+         */
+        val resourceType: String
```

Comment:
nit. we may want to have constants here for known resource types.
I guess we still want `string` instead of enum for unknown types but it would be good to use constants instead of hardcoing strings like "logcat" on the call site.

e.g.

```
data class TestCaseArtifact ( ... ) {
   companion object ResourceTypes {
          val LOCGAT = "logcat" 
   }
}
```

alternatively, we can consider having a ResourceType class and have a way to handle unknown types.

e.g.

```
sealed class ResourceType(val extension: String) {
    data object LOGCAT : ResourceType("logcat")
    data object SCREENSHOT_PROTO : ResourceType("....")
    class Unkonwn(val extension: String) : ResourceType(extension)
   /// maybe implement equals here for `extension` comparison etc
}
```

so we can use `==` for comparisons of these types etc. I htink i like the `RosourceType` solution better. or `TestCaseArtifact.ResourceType` is probably better for scoping.

Reply:
Added companion object ResourceType to TestArtifact class with const strings.